ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -1104,45 +1104,51 @@ def _entries_to_ldif(entries):
     return io.getvalue()
 
 
-def wait_for_replication(ldap, timeout=30):
-    """Wait until updates on all replication agreements are done (or failed)
+def wait_for_replication(ldap, timeout=30,
+                         target_status_re=r'^0 |^Error \(0\) ',
+                         raise_on_timeout=False):
+    """Wait for all replication agreements to reach desired state
 
+    With defaults waits until updates on all replication agreements are
+    done (or failed) and exits without exception
     :param ldap: LDAP client
         autenticated with necessary rights to read the mapping tree
     :param timeout: Maximum time to wait, in seconds
+    :param target_status_re Regexp of status to wait for
+    :param raise_on_timeout if True, raises AssertionError if status not
```

Comment:
Small nit pick, you are missing a ``:`` after the parameter name for the two new optons. It's ``:param <name>: <description>``.